#|_________________Partitions_sets___________________________________________|

##__rule_bwa_index___________________________________\\

bwa_index_partition = config["bwa_index_partition"]
bwa_index_cpus = config["bwa_index_cpus"]
bwa_index_mem = config["bwa_index_mem"]

##__rule_BWA_samblaster_sambamba___________________________\\

BWA_samblaster_sambamba_partition = config["BWA_samblaster_sambamba_partition"]
BWA_samblaster_sambamba_threads = config["BWA_samblaster_sambamba_threads"]
BWA_samblaster_sambamba_memory = config["BWA_samblaster_sambamba_memory"]

##_______rule_picard_dict___________________________\\

picard_dict_partition = config["picard_dict_partition"]
picard_dict_cpus = config["picard_dict_cpus"]
picard_dict_mem = config["picard_dict_mem"]

##______rule_samtools_index__________________________\\

samtools_index_partition = config["samtools_index_partition"]
samtools_index_cpus = config["samtools_index_cpus"]
samtools_index_mem = config["samtools_index_mem"]



#|_________________Threading_sets____________________________________________|

bwa_samblaster_sambamba_threads = config["BWA_samblaster_sambamba_threads"]
call_variants_threads_all_chr = config["Call_Variants_threads_all_chr"]
call_variants_threads_chr1 = config["Call_Variants_threads_chr1"]
call_variants_threads_chr2 = config["Call_Variants_threads_chr2"]
call_variants_threads_chr3 = config["Call_Variants_threads_chr3"]
call_variants_threads_chr4 = config["Call_Variants_threads_chr4"]
call_variants_threads_chr5 = config["Call_Variants_threads_chr5"]
call_variants_threads_chr6 = config["Call_Variants_threads_chr6"]
call_variants_threads_chr7 = config["Call_Variants_threads_chr7"]
call_variants_threads_chr8 = config["Call_Variants_threads_chr8"]
call_variants_threads_chr9 = config["Call_Variants_threads_chr9"]
call_variants_threads_chr10 = config["Call_Variants_threads_chr10"]
call_variants_threads_chr11 = config["Call_Variants_threads_chr11"]
call_variants_threads_chr12 = config["Call_Variants_threads_chr12"]
call_variants_threads_chr13 = config["Call_Variants_threads_chr13"]
call_variants_threads_chr14 = config["Call_Variants_threads_chr14"]
call_variants_threads_chr15 = config["Call_Variants_threads_chr15"]
call_variants_threads_chr16 = config["Call_Variants_threads_chr16"]
call_variants_threads_chr17 = config["Call_Variants_threads_chr17"]
call_variants_threads_chr18 = config["Call_Variants_threads_chr18"]
call_variants_threads_chr19 = config["Call_Variants_threads_chr19"]
call_variants_threads_chr20 = config["Call_Variants_threads_chr20"]
call_variants_threads_chr21 = config["Call_Variants_threads_chr21"]
call_variants_threads_chr22 = config["Call_Variants_threads_chr22"]
call_variants_threads_chrX = config["Call_Variants_threads_chrX"]
call_variants_threads_chrY = config["Call_Variants_threads_chrY"]


#|_________Memory_sets:____________________________________________________________|

bwa_samblaster_sambamba_memory = config["BWA_samblaster_sambamba_memory"]
java_memory = config["Java_memory"]
java_memory_all_chr = config["Java_memory_all_chr"]
java_memory_chr1 = config["Java_memory_chr1"]
java_memory_chr2 = config["Java_memory_chr2"]
java_memory_chr3 = config["Java_memory_chr3"]
java_memory_chr4 = config["Java_memory_chr4"]
java_memory_chr5 = config["Java_memory_chr5"]
java_memory_chr6 = config["Java_memory_chr6"]
java_memory_chr7 = config["Java_memory_chr7"]
java_memory_chr8 = config["Java_memory_chr8"]
java_memory_chr9 = config["Java_memory_chr9"]
java_memory_chr10 = config["Java_memory_chr10"]
java_memory_chr11 = config["Java_memory_chr11"]
java_memory_chr12 = config["Java_memory_chr12"]
java_memory_chr13 = config["Java_memory_chr13"]
java_memory_chr14 = config["Java_memory_chr14"]
java_memory_chr15 = config["Java_memory_chr15"]
java_memory_chr16 = config["Java_memory_chr16"]
java_memory_chr17 = config["Java_memory_chr17"]
java_memory_chr18 = config["Java_memory_chr18"]
java_memory_chr19 = config["Java_memory_chr19"]
java_memory_chr20 = config["Java_memory_chr20"]
java_memory_chr21 = config["Java_memory_chr21"]
java_memory_chr22 = config["Java_memory_chr22"]
java_memory_chrX = config["Java_memory_chrX"]
java_memory_chrY = config["Java_memory_chrY"]


#|__________Dir_settings:____________|

base_dir = config["base_dir"]
main_dir = config["main_dir"] 
genome_dir = config["genome_dir"]
sample_dir = config["sample_dir"] 
output_dir = config["output_dir"] 
logs_dir = config["logs_dir"]

GENOME = config["GENOME"]
KS1 = config["KS1"]
KS2 = config["KS2"]
KS3 = config["KS3"]
KS4 = config["KS4"]
KS5 = config["KS5"]
KS6 = config["KS6"]
KS7 = config["KS7"]

# Genome target files
TARGET_all_chr = config["TARGET_all_chr"]
TARGET_chr1 = config["TARGET_chr1"]
TARGET_chr2 = config["TARGET_chr2"]
TARGET_chr3 = config["TARGET_chr3"]
TARGET_chr4 = config["TARGET_chr4"]
TARGET_chr5 = config["TARGET_chr5"]
TARGET_chr6 = config["TARGET_chr6"]
TARGET_chr7 = config["TARGET_chr7"]
TARGET_chr8 = config["TARGET_chr8"]
TARGET_chr9 = config["TARGET_chr9"]
TARGET_chr10 = config["TARGET_chr10"]
TARGET_chr11 = config["TARGET_chr11"]
TARGET_chr12 = config["TARGET_chr12"]
TARGET_chr13 = config["TARGET_chr13"]
TARGET_chr14 = config["TARGET_chr14"]
TARGET_chr15 = config["TARGET_chr15"]
TARGET_chr16 = config["TARGET_chr16"]
TARGET_chr17 = config["TARGET_chr17"]
TARGET_chr18 = config["TARGET_chr18"]
TARGET_chr19 = config["TARGET_chr19"]
TARGET_chr20 = config["TARGET_chr20"]
TARGET_chr21 = config["TARGET_chr21"]
TARGET_chr22 = config["TARGET_chr22"]
TARGET_chrX = config["TARGET_chrX"]
TARGET_chrY = config["TARGET_chrY"]


# GATK Tools
GATK = config["GATK"]


#|_______________Rules_________________________________________________________|

rule bwa_index: #Create index, fasta.sa
	input:
		fa = main_dir + genome_dir + GENOME + ".fasta"
	output:
		main_dir + genome_dir + GENOME + '.fasta.sa'
	message:
		'Building index -> {output}'
  params: partition = bwa_index_partition
  resources: cpus= bwa_index_cpus, mem_mb= bwa_index_mem
	shell:
		'bwa index {input.fa}' 
		
		
rule BWA_samblaster_sambamba: #Align reads to reference (using BWA); 
	input:
		index = main_dir + genome_dir + GENOME + '.fasta.sa',
		ref = main_dir + genome_dir + GENOME + '.fasta',
		r_fow = main_dir + sample_dir + '{sample}.read1.fastq.gz', #R1
		r_rev = main_dir + sample_dir + '{sample}.read2.fastq.gz'  #R2
	output:
		align = main_dir + output_dir + '{sample}.alignment.bam'
	message:
		'Creating Alignment -> {output}'
  params: partition = BWA_samblaster_sambamba_partition  
	resources: cpus= BWA_samblaster_sambamba_threads, mem_mb= bwa_samblaster_sambamba_memory 
	shell: 
		'bwa mem -t {resources.cpus} -R "@RG\\tID:{wildcards.sample}\\tPL:ILLUMINA\\tSM:{wildcards.sample}" -M {input.ref} {input.r_fow} {input.r_rev} | '
		'samblaster -M |'
		'sambamba view -S -f bam -l 0 /dev/stdin |'
		'sambamba sort -m {resources.mem_mb}MB -t {resources.cpus} -o {output.align} /dev/stdin 2>&1'
		
		
		
rule picard_dict:
	input:
		main_dir + genome_dir + GENOME + '.fasta'
	output:
		main_dir + genome_dir + GENOME + '.dict'
	message:
		'Creating Dictionary with Picard -> {output}'
  params: partition = picard_dict_partition
  resources: cpus= picard_dict_cpus, mem_mb= picard_dict_mem 
	shell:
		'picard CreateSequenceDictionary R={input} O={output}'
		
		
rule samtools_index:
	input:
		main_dir + genome_dir + GENOME + '.fasta'
	output:
		main_dir + genome_dir + GENOME + '.fasta.fai'
	message:
		'Creating index genome fasta file -> {output}'
  params: partition = samtools_index_partition
  resources: cpus= samtools_index_cpus, mem_mb= samtools_index_mem
	shell:
		'samtools faidx {input}'
		
		
"""
# Without Parallelization_______________________________________________________________|

rule gatk_BaseRecalibrator1: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_all_chr
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal.qsub'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'BaseRecalibrator 1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_all_chr
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2.qsub'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'BaseRecalibrator 2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule Analyze_Covariates:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots.pdf'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'Analyzing Covariates -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Apply_BQSR: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal.qsub'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'Applying BQSR -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Call_Variants:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_all_chr
	output:
		site = main_dir + output_dir + '{sample}.gvcf.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars.qsub'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'Processing Call Variants -> {output}'
	threads: call_variants_threads
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
        -L {input.target}' + ' > {output.logs}' 
		
		
rule Joint_Genotyping:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf.gz'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'Processing Joint Genotyping -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Variant_Recalibrator_SNPs:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP.tranches'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'Processing Variant Recalibration for SNPs-> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        	--resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Applying_Recalibration_SNPs:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels.vcf'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'Applying Recalibration for SNPs -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Variant_Recalibrator_Indels:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots.R'
	resources:
        	mem_mb= java_memory_all_chr
	message:
		'Processing Variant Recalibration for Indels -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Applying_Recalibration_Indels:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL.tranches'
	resources:
        	mem_mb= java_memory_all_chr
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf.gz'
	message:
		'Applying Recalibration for Indels -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
"""
		
		
##_______Target_Genome_Parallelization_____________________________________________________________|
		
###__________BaseRecalibrator1_________________________________#
		
rule gatk_BaseRecalibrator1_chr1: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr1
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr1.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr1.qsub'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Processing the first Base Recalibration for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr2: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr2
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr2.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr2.qsub'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Processing the first Base Recalibration for chr2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr3: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr3
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr3.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr3.qsub'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Processing the first Base Recalibration for chr3 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr4: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr4
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr4.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr4.qsub'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Processing the first Base Recalibration for chr4 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr5: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr5
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr5.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr5.qsub'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Processing the first Base Recalibration for chr5 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr6: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr6
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr6.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr6.qsub'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Processing the first Base Recalibration for chr6 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr7: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr7
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr7.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr7.qsub'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Processing the first Base Recalibration for chr7 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr8: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr8
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr8.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr8.qsub'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Processing the first Base Recalibration for chr8 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr9: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr9
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr9.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr9.qsub'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Processing the first Base Recalibration for chr9 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr10: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr10
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr10.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr10.qsub'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Processing the first Base Recalibration for chr10 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr11: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr11
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr11.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr11.qsub'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Processing the first Base Recalibration for chr11 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr12: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr12
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr12.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr12.qsub'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Processing the first Base Recalibration for chr12 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr13: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr13
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr13.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr13.qsub'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Processing the first Base Recalibration for chr13 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr14: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr14
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr14.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr14.qsub'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Processing the first Base Recalibration for chr14 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr15: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr15
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr15.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr15.qsub'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Processing the first Base Recalibration for chr15 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr16: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr16
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr16.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr16.qsub'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Processing the first Base Recalibration for chr16 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr17: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr17
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr17.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr17.qsub'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Processing the first Base Recalibration for chr17 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr18: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr18
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr18.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr18.qsub'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Processing the first Base Recalibration for chr18 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr19: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr19
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr19.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr19.qsub'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Processing the first Base Recalibration for chr19 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr20: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr20
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr20.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr20.qsub'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Processing the first Base Recalibration for chr20 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr21: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr21
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr21.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr21.qsub'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Processing the first Base Recalibration for chr21 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chr22: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chr22
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chr22.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chr22.qsub'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Processing the first Base Recalibration for chr22 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chrX: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chrX
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chrX.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chrX.qsub'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Processing the first Base Recalibration for chrX -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
rule gatk_BaseRecalibrator1_chrY: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bam = main_dir + output_dir + '{sample}.alignment.bam',
		target = TARGET_chrY
	output:
		recal_data = main_dir + output_dir + '{sample}.sorted.bam_pre_recal_data_chrY.grp',
		base_recal = main_dir + output_dir + logs_dir + '{sample}.base_recal_chrY.qsub'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Processing the first Base Recalibration for chrY -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator -R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-I {input.bam} \
		-O {output.recal_data} \
		--interval-padding 100 -L {input.target} ' + '> {output.base_recal}'
		
		
###__________BaseRecalibrator2_________________________________#
		
		
rule gatk_BaseRecalibrator2_chr1: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr1.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr1
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr1.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr1.qsub'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Processing the second Base Recalibration for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr2: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr2.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr2
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr2.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr2.qsub'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Processing the second Base Recalibration for chr2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr3: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr3.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr3
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr3.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr3.qsub'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Processing the second Base Recalibration for chr3 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr4: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr4.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr4
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr4.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr4.qsub'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Processing the second Base Recalibration for chr4 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr5: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr5.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr5
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr5.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr5.qsub'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Processing the second Base Recalibration for chr5 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr6: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr6.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr6
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr6.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr6.qsub'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Processing the second Base Recalibration for chr6 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr7: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr7.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr7
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr7.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr7.qsub'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Processing the second Base Recalibration for chr7 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr8: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr8.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr8
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr8.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr8.qsub'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Processing the second Base Recalibration for chr8 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr9: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr9.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr9
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr9.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr9.qsub'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Processing the second Base Recalibration for chr9 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr10: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr10.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr10
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr10.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr10.qsub'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Processing the second Base Recalibration for chr10 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
rule gatk_BaseRecalibrator2_chr11: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr11.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr11
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr11.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr11.qsub'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Processing the second Base Recalibration for chr11 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr12: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr12.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr12
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr12.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr12.qsub'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Processing the second Base Recalibration for chr12 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr13: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr13.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr13
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr13.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr13.qsub'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Processing the second Base Recalibration for chr13 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr14: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr14.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr14
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr14.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr14.qsub'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Processing the second Base Recalibration for chr14 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr15: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr15.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr15
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr15.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr15.qsub'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Processing the second Base Recalibration for chr15 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr16: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr1.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr16
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr16.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr16.qsub'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Processing the second Base Recalibration for chr16 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr17: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr17.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr17
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr17.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr17.qsub'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Processing the second Base Recalibration for chr17 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr18: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr1.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr18
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr18.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr18.qsub'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Processing the second Base Recalibration for chr18 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr19: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr1.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr19
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr19.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr19.qsub'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Processing the second Base Recalibration for chr19 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr20: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr20.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr20
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr20.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr20.qsub'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Processing the second Base Recalibration for chr20 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr21: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr21.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr21
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr21.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr21.qsub'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Processing the second Base Recalibration for chr21 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chr22: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr22.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chr22
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr22.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chr22.qsub'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Processing the second Base Recalibration for chr22 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chrX: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chrX.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chrX
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chrX.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chrX.qsub'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Processing the second Base Recalibration for chrX -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
rule gatk_BaseRecalibrator2_chrY: 
	input:
		jar = GATK,
		ref = main_dir + genome_dir + GENOME + '.fasta',
		fai = main_dir + genome_dir + GENOME + '.fasta.fai',
		dict = main_dir + genome_dir + GENOME + '.dict',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2,
		ksites3 = main_dir + genome_dir + KS3,
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chrY.grp',
		bam = main_dir + '{sample}.alignment.bam',
		target = TARGET_chrY
	output:
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chrY.grp',
		base_recal = main_dir + output_dir + logs_dir +'{sample}.base_recal_2_chrY.qsub'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Processing the second Base Recalibration for chrY -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} BaseRecalibrator \
		-R {input.ref} \
		--known-sites {input.ksites1} \
		--known-sites {input.ksites2} \
		--known-sites {input.ksites3} \
		-bqsr {input.bqsr} \
		-I {input.bam} \
		-O {output.pos_recal_data} --interval-padding 100 -L {input.target}' + ' > {output.base_recal}'
		
		
###__________Analyze_Covariates_________________________________#
		
		
rule Analyze_Covariates_chr1:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr1.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr1.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr1.pdf'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Analyzing Covariates for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr2:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr2.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr2.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr2.pdf'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Analyzing Covariates for chr2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr3:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr3.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr3.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr3.pdf'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Analyzing Covariates for chr3 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr4:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr4.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr4.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr4.pdf'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Analyzing Covariates for chr4 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr5:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr5.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr5.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr5.pdf'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Analyzing Covariates for chr5 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr6:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr6.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr6.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr6.pdf'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Analyzing Covariates for chr6 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr7:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr7.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr7.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr7.pdf'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Analyzing Covariates for chr7 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr8:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr8.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr8.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr8.pdf'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Analyzing Covariates for chr8 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr9:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr9.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr9.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr9.pdf'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Analyzing Covariates for chr9 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr10:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr10.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr10.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr10.pdf'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Analyzing Covariates for chr10 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr11:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr11.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr11.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr11.pdf'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Analyzing Covariates for chr11 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr12:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr12.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr12.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr12.pdf'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Analyzing Covariates for chr12 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr13:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr13.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr13.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr13.pdf'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Analyzing Covariates for chr13 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr14:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr14.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr14.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr14.pdf'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Analyzing Covariates for chr14 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr15:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr15.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr15.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr15.pdf'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Analyzing Covariates for chr15 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
		
rule Analyze_Covariates_chr16:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr16.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr16.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr16.pdf'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Analyzing Covariates for chr16 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
		
rule Analyze_Covariates_chr17:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr17.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr17.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr17.pdf'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Analyzing Covariates for chr17 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr18:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr18.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr18.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr18.pdf'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Analyzing Covariates for chr18 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr19:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr19.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr19.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr19.pdf'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Analyzing Covariates for chr19 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr20:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr20.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr20.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr20.pdf'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Analyzing Covariates for chr20 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr21:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr21.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr21.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr21.pdf'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Analyzing Covariates for chr21 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chr22:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr22.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chr22.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr22.pdf'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Analyzing Covariates for chr22 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chrX:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chrX.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chrX.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chrX.pdf'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Analyzing Covariates for chrX -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
rule Analyze_Covariates_chrY:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chrY.grp',
		pos_recal_data = main_dir + output_dir +'{sample}.sorted.bam_pos_recal_data_chrY.grp'
	output:
		pdf_log = main_dir + output_dir + logs_dir + '{sample}.sorted.bam_plots_chr1.pdf'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Analyzing Covariates for chrY -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} AnalyzeCovariates \
		-R {input.fa} \
		-before {input.bqsr} \
		-after {input.pos_recal_data} \
		-plots {output.pdf_log}'
		
		
###__________Apply_Base_Quality_Score_Recalibration_________________________________#
		
		
rule Apply_BQSR_chr1: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr1.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr1.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr1.qsub'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Applying Base Quality Score Recalibration for chr1  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr2: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr2.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr2.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr2.qsub'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Applying Base Quality Score Recalibration for chr2  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr3: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr3.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr3.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr3.qsub'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Applying Base Quality Score Recalibration for chr3  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr4: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr4.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr4.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr4.qsub'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Applying Base Quality Score Recalibration for chr4  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr5: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr5.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr5.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr5.qsub'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Applying Base Quality Score Recalibration for chr5  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr6: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr6.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr6.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr6.qsub'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Applying Base Quality Score Recalibration for chr6  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr7: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr7.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr7.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr7.qsub'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Applying Base Quality Score Recalibration for chr7  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr8: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr8.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr8.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr8.qsub'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Applying Base Quality Score Recalibration for chr8  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr9: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr9.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr9.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr9.qsub'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Applying Base Quality Score Recalibration for chr9  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr10: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr1.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr10.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr10.qsub'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Applying Base Quality Score Recalibration for chr10  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr11: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr11.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr11.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr11.qsub'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Applying Base Quality Score Recalibration for chr11  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr12: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr12.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr12.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr12.qsub'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Applying Base Quality Score Recalibration for chr12  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr13: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr13.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr13.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr13.qsub'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Applying Base Quality Score Recalibration for chr13  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr14: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr14.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr14.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr14.qsub'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Applying Base Quality Score Recalibration for chr14  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr15: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr15.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr15.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr15.qsub'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Applying Base Quality Score Recalibration for chr15  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr16: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr16.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr16.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr16.qsub'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Applying Base Quality Score Recalibration for chr16  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr17: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr17.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr17.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr17.qsub'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Applying Base Quality Score Recalibration for chr17  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr18: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr18.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr18.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr18.qsub'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Applying Base Quality Score Recalibration for chr18  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr19: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr19.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr19.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr19.qsub'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Applying Base Quality Score Recalibration for chr19  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr20: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr20.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr20.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr20.qsub'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Applying Base Quality Score Recalibration for chr20  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr21: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr21.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr21.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr21.qsub'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Applying Base Quality Score Recalibration for chr21  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chr22: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chr22.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr22.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chr22.qsub'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Applying Base Quality Score Recalibration for chr22  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chrX: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chrX.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chrX.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chrX.qsub'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Applying Base Quality Score Recalibration for chrX  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
rule Apply_BQSR_chrY: 
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		bqsr = main_dir + output_dir +'{sample}.sorted.bam_pre_recal_data_chrY.grp',
		bam = main_dir + output_dir + '{sample}.alignment.bam',
	output:
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chrY.bam',
		apply_recal = main_dir + output_dir + logs_dir + '{sample}.apply_recal_chrY.qsub'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Applying Base Quality Score Recalibration for chrY  -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyBQSR \
		-R {input.fa} \
		-I {input.bam} \
		--bqsr-recal-file {input.bqsr} \
		-O {output.recal} \
		> {output.apply_recal}'
		
		
###__________Haplotype_Caller_________________________________#
		
		
rule Call_Variants_chr1:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr1.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr1
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr1.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr1.qsub'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Processing Haplotype Caller for chr1 -> {output}'
	threads: call_variants_threads_chr1
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr2:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr2.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr2
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr2.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr2.qsub'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Processing Haplotype Caller for chr2 -> {output}'
	threads: call_variants_threads_chr2
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr3:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr3.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr3
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr3.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr3.qsub'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Processing Haplotype Caller for chr3 -> {output}'
	threads: call_variants_threads_chr3
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr4:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr4.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr4
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr4.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr4.qsub'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Processing Haplotype Caller for chr4 -> {output}'
	threads: call_variants_threads_chr4
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr5:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr5.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr5
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr5.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr5.qsub'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Processing Haplotype Caller for chr5 -> {output}'
	threads: call_variants_threads_chr5
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr6:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr6.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr6
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr6.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr6.qsub'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Processing Haplotype Caller for chr6 -> {output}'
	threads: call_variants_threads_chr6
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr7:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr7.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr7
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr7.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr7.qsub'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Processing Haplotype Caller for chr7 -> {output}'
	threads: call_variants_threads_chr7
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr8:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr8.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr8
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr8.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr8.qsub'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Processing Haplotype Caller for chr8 -> {output}'
	threads: call_variants_threads_chr8
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr9:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr9.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr9
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr9.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr9.qsub'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Processing Haplotype Caller for chr9 -> {output}'
	threads: call_variants_threads_chr9
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr10:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr10.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr10
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr10.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr10.qsub'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Processing Haplotype Caller for chr10 -> {output}'
	threads: call_variants_threads_chr10
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr11:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr11.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr11
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr11.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr11.qsub'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Processing Haplotype Caller for chr11 -> {output}'
	threads: call_variants_threads_chr11
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr12:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr12.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr12
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr12.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr12.qsub'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Processing Haplotype Caller for chr12 -> {output}'
	threads: call_variants_threads_chr12
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr13:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr13.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr13
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr13.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr13.qsub'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Processing Haplotype Caller for chr13 -> {output}'
	threads: call_variants_threads_chr13
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr14:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr14.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr14
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr14.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr14.qsub'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Processing Haplotype Caller for chr14 -> {output}'
	threads: call_variants_threads_chr14
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr15:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr15.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr15
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr15.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr15.qsub'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Processing Haplotype Caller for chr15 -> {output}'
	threads: call_variants_threads_chr15
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr16:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr16.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr16
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr16.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr16.qsub'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Processing Haplotype Caller for chr16 -> {output}'
	threads: call_variants_threads_chr16
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr17:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr17.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr17
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr17.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr17.qsub'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Processing Haplotype Caller for chr17 -> {output}'
	threads: call_variants_threads_chr17
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr18:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr18.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr18
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr18.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr18.qsub'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Processing Haplotype Caller for chr18 -> {output}'
	threads: call_variants_threads_chr18
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr19:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr19.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr19
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr19.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr19.qsub'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Processing Haplotype Caller for chr19 -> {output}'
	threads: call_variants_threads_chr19
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr20:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr20.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr20
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr20.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr20.qsub'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Processing Haplotype Caller for chr20 -> {output}'
	threads: call_variants_threads_chr20
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr21:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr21.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr21
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr21.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr21.qsub'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Processing Haplotype Caller for chr21 -> {output}'
	threads: call_variants_threads_chr21
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chr22:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chr22.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chr22
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chr22.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chr22.qsub'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Processing Haplotype Caller for chr22 -> {output}'
	threads: call_variants_threads_chr22
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chrX:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chrX.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chrX
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chrX.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chrX.qsub'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Processing Haplotype Caller for chrX -> {output}'
	threads: call_variants_threads_chrX
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
rule Call_Variants_chrY:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		recal = main_dir + output_dir + '{sample}.sorted.nodups.realigned.recal_chrY.bam',
		ksites1 = main_dir + genome_dir + KS1,
		target = TARGET_chrY
	output:
		site = main_dir + output_dir + '{sample}.gvcf_chrY.gz',
		logs = main_dir + output_dir + logs_dir + '{sample}.call_vars_chrY.qsub'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Processing Haplotype Caller for chrY -> {output}'
	threads: call_variants_threads_chrY
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} HaplotypeCaller \
		-R {input.fa} \
		-I {input.recal} \
		-D {input.ksites1} \
		-ip 100 -stand-call-conf 10 --emit-ref-confidence GVCF \
		--native-pair-hmm-threads {threads} \
		-O {output.site} \
		-L {input.target}' + ' > {output.logs}'
		
		
###__________Haplotype_Caller_________________________________#
		
		      
rule Joint_Genotyping_chr1:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr1.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr1.gz'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Processing Joint Genotyping for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr2:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr2.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr2.gz'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Processing Joint Genotyping for chr2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr3:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr3.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr3.gz'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Processing Joint Genotyping for chr3 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr4:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr4.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr4.gz'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Processing Joint Genotyping for chr4 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr5:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr5.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr5.gz'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Processing Joint Genotyping for chr 5-> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr6:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr6.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr6.gz'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Processing Joint Genotyping for chr6 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr7:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr7.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr7.gz'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Processing Joint Genotyping for chr7 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr8:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr8.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr8.gz'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Processing Joint Genotyping for chr8 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr9:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr9.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr9.gz'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Processing Joint Genotyping for chr9 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr10:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr10.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr10.gz'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Processing Joint Genotyping for chr10 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr11:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr11.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr11.gz'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Processing Joint Genotyping for chr11 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr12:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr12.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr12.gz'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Processing Joint Genotyping for chr12 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr13:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr13.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr13.gz'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Processing Joint Genotyping for chr13 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr14:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr14.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr14.gz'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Processing Joint Genotyping for chr14 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr15:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr15.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr15.gz'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Processing Joint Genotyping for chr15 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr16:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr16.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr16.gz'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Processing Joint Genotyping for chr16 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr17:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr17.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr17.gz'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Processing Joint Genotyping for chr17 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr18:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr18.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr18.gz'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Processing Joint Genotyping for chr18 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr19:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr19.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr19.gz'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Processing Joint Genotyping for chr19 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr20:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr20.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr20.gz'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Processing Joint Genotyping for chr20 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr21:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr21.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr21.gz'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Processing Joint Genotyping for chr21 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chr22:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chr22.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chr22.gz'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Processing Joint Genotyping for chr22 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chrX:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chrX.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chrX.gz'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Processing Joint Genotyping for chrX -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
rule Joint_Genotyping_chrY:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		site = main_dir + output_dir + '{sample}.gvcf_chrY.gz',
		ksites4 = main_dir + genome_dir + KS4
	output:
		vcf = main_dir + output_dir + '{sample}.vcf_chrY.gz'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Processing Joint Genotyping for chrY -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} GenotypeGVCFs \
		-R {input.fa} --max-alternate-alleles 5 \
		-V {input.site} \
		-O {output.vcf} --dbsnp {input.ksites4} -ip 100'
		
		
###_______________________________Merge_VCFs_____________________________________________________#
		
		
rule Merge_VCFs:
	input:
		vcf_chr1 = main_dir + output_dir + '{sample}.vcf_chr1.gz',
		vcf_chr2 = main_dir + output_dir + '{sample}.vcf_chr2.gz',
		vcf_chr3 = main_dir + output_dir + '{sample}.vcf_chr3.gz',
		vcf_chr4 = main_dir + output_dir + '{sample}.vcf_chr4.gz',
		vcf_chr5 = main_dir + output_dir + '{sample}.vcf_chr5.gz',
		vcf_chr6 = main_dir + output_dir + '{sample}.vcf_chr6.gz',
		vcf_chr7 = main_dir + output_dir + '{sample}.vcf_chr7.gz',
		vcf_chr8 = main_dir + output_dir + '{sample}.vcf_chr8.gz',
		vcf_chr9 = main_dir + output_dir + '{sample}.vcf_chr9.gz',
		vcf_chr10 = main_dir + output_dir + '{sample}.vcf_chr10.gz',
		vcf_chr11 = main_dir + output_dir + '{sample}.vcf_chr11.gz',
		vcf_chr12 = main_dir + output_dir + '{sample}.vcf_chr12.gz',
		vcf_chr13 = main_dir + output_dir + '{sample}.vcf_chr13.gz',
		vcf_chr14 = main_dir + output_dir + '{sample}.vcf_chr14.gz',
		vcf_chr15 = main_dir + output_dir + '{sample}.vcf_chr15.gz',
		vcf_chr16 = main_dir + output_dir + '{sample}.vcf_chr16.gz',
		vcf_chr17 = main_dir + output_dir + '{sample}.vcf_chr17.gz',
		vcf_chr18 = main_dir + output_dir + '{sample}.vcf_chr18.gz',
		vcf_chr19 = main_dir + output_dir + '{sample}.vcf_chr19.gz',
		vcf_chr20 = main_dir + output_dir + '{sample}.vcf_chr20.gz',
		vcf_chr21 = main_dir + output_dir + '{sample}.vcf_chr21.gz',
		vcf_chr22 = main_dir + output_dir + '{sample}.vcf_chr22.gz',
		vcf_chrX = main_dir + output_dir + '{sample}.vcf_chrX.gz',
		vcf_chrY = main_dir + output_dir + '{sample}.vcf_chrY.gz'
	output:
		merged_vcfs = main_dir + output_dir + '{sample}_vcf.gz'
	message:
		'Merging all Vcfs -> {output}'
	shell:
		'picard MergeVcfs \
		I={input.vcf_chr1} I={input.vcf_chr2} I={input.vcf_chr3} I={input.vcf_chr4} I={input.vcf_chr5} \
		I={input.vcf_chr6} I={input.vcf_chr7} I={input.vcf_chr8} I={input.vcf_chr9} I={input.vcf_chr10} I={input.vcf_chr11} \
		I={input.vcf_chr12} I={input.vcf_chr13} I={input.vcf_chr14} I={input.vcf_chr15} I={input.vcf_chr16} I={input.vcf_chr17} \
		I={input.vcf_chr18} I={input.vcf_chr19} I={input.vcf_chr20} I={input.vcf_chr21} I={input.vcf_chr22} I={input.vcf_chrX} I={input.vcf_chrY} \
		O={output.merged_vcfs}'
		
		
###__________Analyze_Covariates_________________________________#
		
		
rule Variant_Recalibrator_SNPs_chr1:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr1.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr1.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr1.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr1.tranches'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Processing Variant Recalibration with SNPs for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr2:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr2.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr2.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr2.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr2.tranches'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Processing Variant Recalibration with SNPs for chr2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr3:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr3.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr3.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr3.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr3.tranches'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Processing Variant Recalibration with SNPs for chr3 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr4:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr4.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr4.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr4.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr4.tranches'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Processing Variant Recalibration with SNPs for chr4 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr5:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr5.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr5.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr5.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr5.tranches'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Processing Variant Recalibration with SNPs for chr5 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr6:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr6.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr6.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr6.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr6.tranches'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Processing Variant Recalibration with SNPs for chr6 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr7:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr7.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr7.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr7.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr7.tranches'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Processing Variant Recalibration with SNPs for chr7 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr8:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr8.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr8.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr8.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr8.tranches'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Processing Variant Recalibration with SNPs for chr8 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr9:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr9.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr9.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr9.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr9.tranches'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Processing Variant Recalibration with SNPs for chr9 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr10:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr10.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr10.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr10.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr10.tranches'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Processing Variant Recalibration with SNPs for chr10 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr11:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr11.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr11.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr11.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr11.tranches'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Processing Variant Recalibration with SNPs for chr11 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr12:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr12.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr12.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr12.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr12.tranches'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Processing Variant Recalibration with SNPs for chr12 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr13:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr13.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr13.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr13.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr13.tranches'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Processing Variant Recalibration with SNPs for chr13 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr14:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr14.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr14.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr14.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr14.tranches'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Processing Variant Recalibration with SNPs for chr14 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr15:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr15.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr15.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr15.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr15.tranches'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Processing Variant Recalibration with SNPs for chr15 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr16:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr16.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr16.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr16.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr16.tranches'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Processing Variant Recalibration with SNPs for chr16 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr17:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr17.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr17.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr17.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr17.tranches'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Processing Variant Recalibration with SNPs for chr17 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr18:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr18.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr18.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr18.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr18.tranches'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Processing Variant Recalibration with SNPs for chr18 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr19:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr19.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr19.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr19.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr19.tranches'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Processing Variant Recalibration with SNPs for chr19 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr20:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr20.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr20.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr20.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr20.tranches'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Processing Variant Recalibration with SNPs for chr20 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr21:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr21.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr21.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr21.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr21.tranches'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Processing Variant Recalibration with SNPs for chr21 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chr22:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr22.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr22.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chr22.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr22.tranches'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Processing Variant Recalibration with SNPs for chr22 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chrX:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chrX.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chrX.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chrX.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chrX.tranches'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Processing Variant Recalibration with SNPs for chrX -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
rule Variant_Recalibrator_SNPs_chrY:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chrY.gz',
		ksites1 = main_dir + genome_dir + KS1,
		ksites3 = main_dir + genome_dir + KS3,
		ksites5 = main_dir + genome_dir + KS5,
		ksites6 = main_dir + genome_dir + KS6
	output:
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chrY.recal',
		rscript_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_plots_chrY.R',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chrY.tranches'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Processing Variant Recalibration with SNPs for chrY -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf} \
		-mode SNP -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 6 -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
		-O {output.recal_snp} \
		--tranches-file {output.tranches_snp} \
		--rscript-file {output.rscript_snp} \
        --resource:hapmap,known=false,training=true,truth=true,prior=15.0 {input.ksites5} \
		--resource:omni,known=false,training=true,truth=true,prior=12.0 {input.ksites6} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1} \
		--resource:1000G,known=false,training=true,truth=false,prior=10.0 {input.ksites3}'
		
		
###__________Applying_Recalibration_SNPs_________________________________#
		
		
rule Applying_Recalibration_SNPs_chr1:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr1.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr1.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr1.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr1.vcf'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Applying Recalibration with SNPs for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr2:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr2.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr2.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr2.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr2.vcf'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Applying Recalibration with SNPs for chr2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr3:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr3.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr3.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr3.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr3.vcf'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Applying Recalibration with SNPs for chr3 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr4:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr4.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr4.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr4.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr4.vcf'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Applying Recalibration with SNPs for chr4 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr5:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr5.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr5.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr5.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr5.vcf'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Applying Recalibration with SNPs for chr5 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr6:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr6.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr6.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr6.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr6.vcf'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Applying Recalibration with SNPs for chr6 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr7:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr7.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr7.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr7.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr7.vcf'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Applying Recalibration with SNPs for chr7 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr8:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr8.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr8.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr8.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr8.vcf'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Applying Recalibration with SNPs for chr8 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr9:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr9.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr9.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr9.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr9.vcf'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Applying Recalibration with SNPs for chr9 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr10:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr10.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr10.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr10.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr10.vcf'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Applying Recalibration with SNPs for chr10 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr11:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr11.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr11.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr11.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr11.vcf'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Applying Recalibration with SNPs for chr11 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr12:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr12.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr12.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr12.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr12.vcf'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Applying Recalibration with SNPs for chr12 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr13:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr13.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr13.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr13.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr13.vcf'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Applying Recalibration with SNPs for chr13 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr14:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr14.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr14.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr14.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr14.vcf'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Applying Recalibration with SNPs for chr14 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr15:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr15.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr15.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr15.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr15.vcf'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Applying Recalibration with SNPs for chr15 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr16:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr16.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr16.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr16.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr16.vcf'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Applying Recalibration with SNPs for chr16 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr17:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr17.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr17.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr17.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr17.vcf'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Applying Recalibration with SNPs for chr17 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr18:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr18.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr18.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr18.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr18.vcf'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Applying Recalibration with SNPs for chr18 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr19:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr19.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr19.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr19.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr19.vcf'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Applying Recalibration with SNPs for chr19 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr20:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr20.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr20.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr20.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr20.vcf'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Applying Recalibration with SNPs for chr20 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr21:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr21.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr21.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr21.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr21.vcf'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Applying Recalibration with SNPs for chr21 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chr22:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chr22.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr22.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chr22.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr22.vcf'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Applying Recalibration with SNPs for chr22 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chrX:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chrX.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chrX.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chrX.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chrX.vcf'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Applying Recalibration with SNPs for chrX -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
rule Applying_Recalibration_SNPs_chrY:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf = main_dir + output_dir + '{sample}.vcf_chrY.gz',
		recal_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chrY.recal',
		tranches_snp = main_dir + output_dir + '{sample}.recalibrate_SNP_chrY.tranches'
	output:
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chrY.vcf'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Applying Recalibration with SNPs for chrY -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf} \
		-O {output.vcf_snp} \
		--recal-file {input.recal_snp} \
		--tranches-file {input.tranches_snp} \
		-truth-sensitivity-filter-level 99.5 \
		--create-output-variant-index true -mode SNP'
		
		
###__________Variant_Recalibrator_Indels_________________________________#
		
		
rule Variant_Recalibrator_Indels_chr1:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr1.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr1.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr1.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr1.R'
	resources:
        	mem_mb= java_memory_chr1
	message:
		'Processing Variant Recalibration with Indels for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr2:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr2.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr2.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr2.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr2.R'
	resources:
        	mem_mb= java_memory_chr2
	message:
		'Processing Variant Recalibration with Indels for chr2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr3:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr3.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr3.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr3.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr3.R'
	resources:
        	mem_mb= java_memory_chr3
	message:
		'Processing Variant Recalibration with Indels for chr3 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr4:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr4.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr4.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr4.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr4.R'
	resources:
        	mem_mb= java_memory_chr4
	message:
		'Processing Variant Recalibration with Indels for chr4 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr5:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr5.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr5.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr5.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr5.R'
	resources:
        	mem_mb= java_memory_chr5
	message:
		'Processing Variant Recalibration with Indels for chr5 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr6:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr6.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr6.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr6.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr6.R'
	resources:
        	mem_mb= java_memory_chr6
	message:
		'Processing Variant Recalibration with Indels for chr6 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr7:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr7.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr7.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr7.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr7.R'
	resources:
        	mem_mb= java_memory_chr7
	message:
		'Processing Variant Recalibration with Indels for chr7 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr8:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr8.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr8.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr8.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr8.R'
	resources:
        	mem_mb= java_memory_chr8
	message:
		'Processing Variant Recalibration with Indels for chr8 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr9:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr9.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr9.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr9.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr9.R'
	resources:
        	mem_mb= java_memory_chr9
	message:
		'Processing Variant Recalibration with Indels for chr9 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr10:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr10.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr10.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr10.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr10.R'
	resources:
        	mem_mb= java_memory_chr10
	message:
		'Processing Variant Recalibration with Indels for chr10 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr11:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr11.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr11.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr11.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr11.R'
	resources:
        	mem_mb= java_memory_chr11
	message:
		'Processing Variant Recalibration with Indels for chr11 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr12:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr12.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr12.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr12.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr12.R'
	resources:
        	mem_mb= java_memory_chr12
	message:
		'Processing Variant Recalibration with Indels for chr12 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr13:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr13.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr13.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr13.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr13.R'
	resources:
        	mem_mb= java_memory_chr13
	message:
		'Processing Variant Recalibration with Indels for chr13 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr14:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr14.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr14.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr14.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr14.R'
	resources:
        	mem_mb= java_memory_chr14
	message:
		'Processing Variant Recalibration with Indels for chr14 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr15:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr15.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr15.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr15.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr15.R'
	resources:
        	mem_mb= java_memory_chr15
	message:
		'Processing Variant Recalibration with Indels for chr15 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr16:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr16.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr16.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr16.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr16.R'
	resources:
        	mem_mb= java_memory_chr16
	message:
		'Processing Variant Recalibration with Indels for chr16 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr17:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr17.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr17.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr17.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr17.R'
	resources:
        	mem_mb= java_memory_chr17
	message:
		'Processing Variant Recalibration with Indels for chr17 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr18:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr18.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr18.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr18.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr1.R'
	resources:
        	mem_mb= java_memory_chr18
	message:
		'Processing Variant Recalibration with Indels for chr18 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr19:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr19.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr19.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr19.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr19.R'
	resources:
        	mem_mb= java_memory_chr19
	message:
		'Processing Variant Recalibration with Indels for chr19 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr20:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr20.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr20.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr20.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr20.R'
	resources:
        	mem_mb= java_memory_chr20
	message:
		'Processing Variant Recalibration with Indels for chr20 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr21:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr21.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr21.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr21.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr21.R'
	resources:
        	mem_mb= java_memory_chr21
	message:
		'Processing Variant Recalibration with Indels for chr21 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chr22:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr22.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr22.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr22.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chr22.R'
	resources:
        	mem_mb= java_memory_chr22
	message:
		'Processing Variant Recalibration with Indels for chr22 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chrX:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chrX.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chrX.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chrX.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chrX.R'
	resources:
        	mem_mb= java_memory_chrX
	message:
		'Processing Variant Recalibration with Indels for chrX -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
rule Variant_Recalibrator_Indels_chrY:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chrY.vcf',
		ksites1 = main_dir + genome_dir + KS1,
		ksites2 = main_dir + genome_dir + KS2
	output:
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chrY.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chrY.tranches',
		rscript_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_plots_chrY.R'
	resources:
        	mem_mb= java_memory_chrY
	message:
		'Processing Variant Recalibration with Indels for chrY -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} VariantRecalibrator \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_indel} \
		--tranches-file {output.tranches_indel} \
		--rscript-file {output.rscript_indel} \
		-mode INDEL -tranche 100.0 -tranche 99.9 -tranche 99.0 -tranche 90.0 \
		--max-gaussians 4 -an QD -an FS -an DP -an MQRankSum -an ReadPosRankSum -an SOR \
		--resource:mills,known=false,training=true,truth=true,prior=12.0 {input.ksites2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 {input.ksites1}'
		
		
###__________Variant_Recalibrator_Indels_________________________________#
		
		
rule Applying_Recalibration_Indels_chr1:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr1.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr1.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr1.tranches'
	resources:
        	mem_mb= java_memory_chr1
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr1.gz'
	message:
		'Applying Recalibration with Indels for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr2:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr2.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr2.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr2.tranches'
	resources:
        	mem_mb= java_memory_chr2
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr2.gz'
	message:
		'Applying Recalibration with Indels for chr2 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr3:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr3.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr3.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr3.tranches'
	resources:
        	mem_mb= java_memory_chr3
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr3.gz'
	message:
		'Applying Recalibration with Indels for chr3 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr4:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr4.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr4.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr4.tranches'
	resources:
        	mem_mb= java_memory_chr4
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr4.gz'
	message:
		'Applying Recalibration with Indels for chr4 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr5:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr5.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr5.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr5.tranches'
	resources:
        	mem_mb= java_memory_chr5
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr5.gz'
	message:
		'Applying Recalibration with Indels for chr5 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr6:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr6.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr6.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr6.tranches'
	resources:
        	mem_mb= java_memory_chr6
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr6.gz'
	message:
		'Applying Recalibration with Indels for chr6 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr7:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr7.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr7.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr7.tranches'
	resources:
        	mem_mb= java_memory_chr7
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr7.gz'
	message:
		'Applying Recalibration with Indels for chr7 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr8:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr8.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr8.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr8.tranches'
	resources:
        	mem_mb= java_memory_chr8
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr8.gz'
	message:
		'Applying Recalibration with Indels for chr8 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr9:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr9.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr9.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr9.tranches'
	resources:
        	mem_mb= java_memory_chr9
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr9.gz'
	message:
		'Applying Recalibration with Indels for chr9 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr10:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr10.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr10.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr10.tranches'
	resources:
        	mem_mb= java_memory_chr10
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr10.gz'
	message:
		'Applying Recalibration with Indels for chr10 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr11:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr11.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr11.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr11.tranches'
	resources:
        	mem_mb= java_memory_chr11
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr11.gz'
	message:
		'Applying Recalibration with Indels for chr11 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr12:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr12.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr12.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr12.tranches'
	resources:
        	mem_mb= java_memory_chr12
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr12.gz'
	message:
		'Applying Recalibration with Indels for chr1 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr13:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr13.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr13.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr13.tranches'
	resources:
        	mem_mb= java_memory_chr13
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr13.gz'
	message:
		'Applying Recalibration with Indels for chr13 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr14:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr14.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr14.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr14.tranches'
	resources:
        	mem_mb= java_memory_chr14
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr14.gz'
	message:
		'Applying Recalibration with Indels for chr14 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr15:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr15.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr15.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr15.tranches'
	resources:
        	mem_mb= java_memory_chr15
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr15.gz'
	message:
		'Applying Recalibration with Indels for chr15 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr16:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr16.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr16.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr16.tranches'
	resources:
        	mem_mb= java_memory_chr16
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr16.gz'
	message:
		'Applying Recalibration with Indels for chr16 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr17:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr17.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr17.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr17.tranches'
	resources:
        	mem_mb= java_memory_chr17
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr17.gz'
	message:
		'Applying Recalibration with Indels for chr17 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr18:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr18.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr18.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr18.tranches'
	resources:
        	mem_mb= java_memory_chr18
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr18.gz'
	message:
		'Applying Recalibration with Indels for chr18 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr19:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr19.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr19.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr19.tranches'
	resources:
        	mem_mb= java_memory_chr19
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr19.gz'
	message:
		'Applying Recalibration with Indels for chr19 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr20:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr20.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr20.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr20.tranches'
	resources:
        	mem_mb= java_memory_chr20
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr20.gz'
	message:
		'Applying Recalibration with Indels for chr20 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr21:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr21.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr21.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr21.tranches'
	resources:
        	mem_mb= java_memory_chr21
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr21.gz'
	message:
		'Applying Recalibration with Indels for chr21 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chr22:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chr22.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr22.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chr22.tranches'
	resources:
        	mem_mb= java_memory_chr22
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr22.gz'
	message:
		'Applying Recalibration with Indels for chr22 -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chrX:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chrX.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chrX.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chrX.tranches'
	resources:
        	mem_mb= java_memory_chrX
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chrX.gz'
	message:
		'Applying Recalibration with Indels for chrX -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
rule Applying_Recalibration_Indels_chrY:
	input:
		jar = GATK,
		fa = main_dir + genome_dir + GENOME + ".fasta",
		vcf_snp = main_dir + output_dir + '{sample}.recalibrated_snps_raw_indels_chrY.vcf',
		recal_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chrY.recal',
		tranches_indel = main_dir + output_dir + '{sample}.recalibrate_INDEL_chrY.tranches'
	resources:
        	mem_mb= java_memory_chrY
	output:
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chrY.gz'
	message:
		'Applying Recalibration with Indels for chrY -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} ApplyVQSR \
		-R {input.fa} \
		-V {input.vcf_snp} \
		-O {output.recal_vcf} \
		--recal-file {input.recal_indel} \
		--tranches-file {input.tranches_indel} \
		-truth-sensitivity-filter-level 99.0 \
		--create-output-variant-index true -mode INDEL'
		
		
###_______________________________Merge_Recalibrated_VCFs____________________________________________________#
		
		
rule Merge_Recalibrated_VCFs:
	input:
		vcf_chr1 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr1.gz',
		vcf_chr2 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr2.gz',
		vcf_chr3 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr3.gz',
		vcf_chr4 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr4.gz',
		vcf_chr5 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr5.gz',
		vcf_chr6 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr6.gz',
		vcf_chr7 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr7.gz',
		vcf_chr8 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr8.gz',
		vcf_chr9 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr9.gz',
		vcf_chr10 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr10.gz',
		vcf_chr11 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr11.gz',
		vcf_chr12 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr12.gz',
		vcf_chr13 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr13.gz',
		vcf_chr14 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr14.gz',
		vcf_chr15 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr15.gz',
		vcf_chr16 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr16.gz',
		vcf_chr17 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr17.gz',
		vcf_chr18 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr18.gz',
		vcf_chr19 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr19.gz',
		vcf_chr20 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr20.gz',
		vcf_chr21 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr21.gz',
		vcf_chr22 = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chr22.gz',
		vcf_chrX = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chrX.gz',
		vcf_chrY = main_dir + output_dir + '{sample}.recalibrated_variants.vcf_chrY.gz'
	output:
		merged_recalibrated_vcfs = main_dir + output_dir + '{sample}.recalibrated_variants.vcf.gz'
	message:
		'Merging all Recalibrated Vcfs -> {output}'
	shell:
		'picard MergeVcfs \
		I={input.vcf_chr1} I={input.vcf_chr2} I={input.vcf_chr3} I={input.vcf_chr4} I={input.vcf_chr5} \
		I={input.vcf_chr6} I={input.vcf_chr7} I={input.vcf_chr8} I={input.vcf_chr9} I={input.vcf_chr10} I={input.vcf_chr11} \
		I={input.vcf_chr12} I={input.vcf_chr13} I={input.vcf_chr14} I={input.vcf_chr15} I={input.vcf_chr16} I={input.vcf_chr17} \
		I={input.vcf_chr18} I={input.vcf_chr19} I={input.vcf_chr20} I={input.vcf_chr21} I={input.vcf_chr22} I={input.vcf_chrX} I={input.vcf_chrY} \
		O={output.merged_recalibrated_vcfs}'
		
		       
###_______________________________Calculate_Genotype_Posteriors____________________________________________________#
		
		
rule Calculate_Genotype_Posteriors:
	input:
		jar = GATK,
		recal_vcf = main_dir + output_dir + '{sample}.recalibrated_variants.vcf.gz',
		ksites7 = main_dir + genome_dir + KS7
	output:
		recal_vcf_CGP = main_dir + output_dir + '{sample}.recalibrated_variants_postCGP.vcf.gz'
	resources:
        	mem_mb= java_memory
	message:
		'Calculating Genotype Posteriors -> {output}'
	shell:
		'java -Xmx{resources.mem_mb}g -jar {input.jar} CalculateGenotypePosteriors \
		-V {input.recal_vcf} \
		-O {output.recal_vcf_CGP} \
		-supporting {input.ksites7}'
